exonum-java-binding-core/src/test/java/com/exonum/binding/storage/indices/ProofListContainsMatcher.java

```diff
@@ -50,11 +52,16 @@ protected boolean matchesSafely(ProofListIndexProxy<String> list) {
     }
 
     ListProof proof = proofFunction.apply(list);
-    ListProofRootHashCalculator<String> calculator = newProofValidator();
+
+    ListProofStructureValidator validator = newProofStructureValidator();
+    proof.accept(validator);
+
+    ListProofRootHashCalculator<String> calculator = newProofHashCodeCalculator();
     proof.accept(calculator);
 
-    //TODO add structure check or root hash check ?
-    return elementsMatcher.matches(calculator.getElements());
+    return validator.check().equals(ListProofStatus.VALID)
```

Comment:
`check` suggests an action, but we just access an already determined value.

`validator.isValid()`? Or `validator.getResult() == ListProofStatus.VALID`?